### PR TITLE
Fix manual address input

### LINKED ISSUE #1416: Fix manual address input for Swap and Bridge
Somewhere during recent refactoring this stopped working
image

### PR DESCRIPTION
**Pull request by hetfly**
close #1416

### CODE REVIEW

File: apps/web/src/hooks/useManualAddresses.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { XChainId } from '@/types';
+import { useCallback, useState } from 'react';
+
+export default function useManualAddresses(): {
+  manualAddresses: { [xChainId: string]: string | undefined };
+  setManualAddress: (xChainId: XChainId, address: string | undefined) => void;
```
Comment (0xmilktea) [inline apps/web/src/hooks/useManualAddresses.tsx:6]: is the hook still useful?

**Review by parrot9design — APPROVED** [review]
Much better. Thanks for taking care of this.

**Comment by parrot9design** [comment]
Actually, I did find an issue when signed in with an ICON, Havah, and/or Archway wallet: If you enter a custom address for, say, BNB Chain, it keeps the custom address for all other EVM chains (expected), but also for Injective. It doesn't register that it's an invalid address for Injective until I click to change it.

**Comment by hetfly** [comment]
Fixed the injective issue, but it was part of the (IMO fault) logic that kept manual chain address across compatible chains. I think it might be better to keep the manually added address only per currently active chain, though, meaning that if I input BNB address, then change to arbitrum, it defaults to logged in wallet, rather than the BNB edited address.

**Comment by parrot9design** [comment]
Thanks @hetfly, I'm happy with that handling.

One thing I found myself wanting during testing is the ability to hit Enter rather than click Use for the custom address. Is it possible to add that? Otherwise this is good to merge.

**Pull request merged**